### PR TITLE
improve SpacingD output shape compute stability

### PR DESCRIPTION
**Pull request by wyli**
for slightly different affine (atol < 1e-3) the shape output might be different
```
sum([1.49999991e+00 1.44128689e-04] * [237. 144.]) = sum([3.55499980e+02 2.07545313e-02]) = 355.52073414989167

sum([ 1.49999989e+00 -6.16167492e-10] * [237. 144.]) = sum([ 3.55499975e+02 -8.87281188e-08]) = 355.4999748785736
```

- increase the stability of affine inv
- check affine floating point types
- ensure same shape when there's only tiny difference in pixdim 

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by myron** [comment]
Looks good. Thanks!

**Review by myron — APPROVED** [review]

File: tests/test_global_mutual_information_loss.py [hunk 1]
```diff
@@ -26,28 +26,28 @@
...
+        -1.5860257,
+        -0.30265224,
+        -0.18666176,
+        -0.15887907,
+        -0.1625064,
+        -0.16603896,
+        -0.19222091,
+        -0.18158069,
+        -0.167644,
+        -0.16698098,
```
Comment (wyli) [inline tests/test_global_mutual_information_loss.py:50]: difference introduced by the resampler floating points--

previous `(2/(d-1)*g+1)*(d-1)/2=g+(d-1)/2` vs now `((2/d*g+1)*d-1)/2=g+(d-1)/2`

**Pull request merged**